### ISSUE TITLE
Data import directly from paste buffer

### ISSUE THREAD

**Issue by philrz** [brimdata/zui#2909]
## tl;dr

Particularly with the coming support for TSV format, Zui users would benefit from being able to import data directly from their paste buffer.

## Details

At the time of the filing of this issue, Zui is at commit 3a66931.

https://github.com/brimdata/zed/pull/4891 will soon bring support to Zed for easily reading TSV data. Once that lands I intend to add TSV to the list of import/export formats in Zui (#2908).

Zui users that have asked for TSV support have primarily been users of Excel or Google Sheets that have pointed out when doing cut & paste from these apps the data in the paste buffer is of TSV format. The attached format shows where things are headed at the Zed layer using the branch from https://github.com/brimdata/zed/pull/4891.

https://github.com/brimdata/zui/assets/5934157/4361e34a-aa84-4ebb-b63b-ecc3f71312b3

As this shows, it'll now be easy to read this TSV data, but the user needs to take the extra step if pasting it into a file. Such a file could be easily dragged into Zui, but even better would be if they could skip the file altogether and go directly from the paste buffer into Zui. I'm hopeful that this will flow nicely with the new Preview & Load experience, i.e., instead of having the `zq` take its input from files on a command line, just give it a stdin stream of what's in the paste buffer.

In terms of how to invoke this, I'm not yet certain. My first thought was to add another pull-down menu item similar to the **Open Data File...** (e.g., "Open From Paste Buffer"?) @jameskerr also wondered about taking over the **Paste** option that's in the **Edit** menu, which does seem like it would be possible in contexts where it's not already claimed for something else (e.g., when the Editor is active).

**Comment by philrz** [comment]
Verified in Zui commit 535f6e9.

As shown in the attached video, data in the paste buffer (such as TSV from Google Sheets) can now be imported directly into Zui using the **File > Paste Data...** option from the pull-down menu or corresponding hot key (e.g., shift-command-v on macOS). Once pasted, it becomes input data that can be shaped in **Preview & Load** just like files dragged into the app. Multiple pastes are possible as well as mixing pastes with file inputs. Another nice trick shown is that this makes it easy to import simple text from the paste buffer if the **Data Format** option is for **Line** is selected.

https://github.com/brimdata/zui/assets/5934157/5504851c-ef2f-4faa-9ed1-183e357a8b10

Thanks @jameskerr!
